pkg/collector/helm_collector.go

```diff
@@ -0,0 +1,61 @@
+package collector
+
+import (
+	"path/filepath"
+
+	"github.com/Azure/aks-periscope/pkg/interfaces"
+	"github.com/Azure/aks-periscope/pkg/utils"
+)
+
+// HelmCollector defines a Helm Collector struct
+type HelmCollector struct {
+	BaseCollector
+}
+
+var _ interfaces.Collector = &HelmCollector{}
+
+// NewHelmCollector is a constructor
+func NewHelmCollector(exporter interfaces.Exporter) *HelmCollector {
+	return &HelmCollector{
+		BaseCollector: BaseCollector{
+			collectorType: Helm,
+			exporter:      exporter,
+		},
+	}
+}
+
+// Collect implements the interface method
+func (collector *HelmCollector) Collect() error {
+	rootPath, err := utils.CreateCollectorDir(collector.GetName())
+	if err != nil {
+		return err
+	}
+
+	helmListFile := filepath.Join(rootPath, "helm_list")
+	output, err := utils.RunCommandOnContainer("helm", "list", "--all-namespaces")
+	if err != nil {
+		return err
+	}
+
+	err = utils.WriteToFile(helmListFile, output)
+	if err != nil {
+		return err
+	}
+
+	collector.AddToCollectorFiles(helmListFile)
+
+	helmHistoryFile := filepath.Join(rootPath, "helm_history")
+	output, err = utils.RunCommandOnContainer("helm", "history", "-n", "default", "azure-arc")
```

Comment:
Rather than adding some packages on the Docker image and executing command lines from our code, should we use libraries instead ? 
https://github.com/helm/helm/blob/0af54d0f5c0f2a108506733b64752039ee39d165/cmd/helm/chart_list.go#L33
https://github.com/helm/helm/blob/0af54d0f5c0f2a108506733b64752039ee39d165/cmd/helm/history.go#L54

That way we can have better error handling and smaller docker image.

Last point: it is best practice to create new variables rather than doing variables reuse (`output, err =` )

Reply:
Thank you @arnaud-tincelin, for taking the time to review my changes, these are very good suggestions. 
Also tagging relevant people: @davidkydd @safeermohammed @Tatsinnit 

Regarding the suggestion about utilizing helm library, but one thing to keep in mind is that we, the arc team, are trying to get an MVP out by end of May (a.k.a. next week) which will include this new helm feature for periscope. Refactoring the helm classes so that it uses library functions seems to be a big change and the changes that I have so far have already been tested extensively.

After discussing offline with the periscope team, we had a suggestion for periscope team to make the collectors use library functions even for the kubectl commands in the code. So the helm collector can be taking care of as part of this refactoring.  

For now, given the short amount of time we have left until the MVP needs to be released, my thinking is to freeze the code changes for the time being. But of course, I will do my best to fix the other smaller changes such as creating the new variables and combining the lines in the Dockerfile. 

Please let me know your thoughts on this. 

Of course,

Reply:
Later refactoring to utilize helm + kubectl from libs rather than invoking executables sounds good to me

Reply:
+1, yes we should do it as part of refactor, I will add a workitem with more details by eod.

I will test any new changes sometime Monday our time or earlier. (Will kee you guys posted)

thanks guys 🙏